.golangci.yml

```diff
@@ -0,0 +1,43 @@
+service:
+  golangci-lint-version: 1.21.0 # Keep this in sync with .circleci/config.yml
+
+run:
+  timeout: 5m
+  modules-download-mode: vendor
+  skip-dirs:
+    - mocks
+
+linters-settings:
+  gofmt:
+    simplify: true
+  govet:
+    check-shadowing: true
+    enable-all: true
+
+linters:
+  disable-all: true
+  enable:
+    - deadcode
+    - gofmt
+    - golint
+    - gosimple
+    - govet
+    - ineffassign
+    - structcheck
+    - unconvert
+    - unused
+    - varcheck
+    - errcheck
+
+issues:
+  exclude-rules:
+    - linters:
+      # ignore unused warnings from enterprise code
+      # add more as required.
+      - unused
+      text: "testLogsCmdF|MmctlE2ETestSuite|MmctlUnitTestSuite|setupTestHelper"
+
+    - linters:
+      # ignore golint error for a lot of packages for now
+      - golint
+      path: "client"
```

Comment:
To keep consistency with the server-side functions. In the future we can change those too